app/models/manageiq/providers/hawkular/concerns/server_operations.rb

```diff
@@ -0,0 +1,87 @@
+module ManageIQ::Providers
+  module Hawkular
+    module ServerOperations
+      # server ops
+      def shutdown_middleware_server(ems_ref, params = {})
+        run_generic_operation(:Shutdown, ems_ref,
+                              :restart => false, :timeout => params[:timeout] || 0)
+      end
+
+      def suspend_middleware_server(ems_ref, params = {})
+        run_generic_operation(:Suspend, ems_ref, :timeout => params[:timeout] || 0)
+      end
+
+      def resume_middleware_server(ems_ref)
+        run_generic_operation(:Resume, ems_ref)
+      end
+
+      def reload_middleware_server(ems_ref)
+        run_generic_operation(:Reload, ems_ref)
+      end
+
+      def stop_middleware_server(ems_ref)
+        run_generic_operation(:Shutdown, ems_ref)
+      end
+
+      def start_middleware_domain_server(ems_ref)
+        run_generic_operation(:Start, ems_ref)
+      end
+
+      def stop_middleware_domain_server(ems_ref)
+        run_generic_operation(:Stop, ems_ref)
+      end
+
+      def restart_middleware_server(ems_ref)
+        run_generic_operation(:Shutdown, ems_ref, :restart => true)
+      end
+
+      # domain server ops
+      def restart_middleware_domain_server(ems_ref)
+        run_generic_operation(:Restart, ems_ref)
+      end
+
+      def kill_middleware_domain_server(ems_ref)
+        run_generic_operation(:Kill, ems_ref)
+      end
+
+      # server group ops
+      def start_middleware_server_group(ems_ref)
+        run_generic_operation('Start Servers', ems_ref)
+      end
+
+      def stop_middleware_server_group(ems_ref, params = {})
+        run_generic_operation('Stop Servers', ems_ref, :timeout => params[:timeout] || 0)
+      end
+
+      def restart_middleware_server_group(ems_ref)
+        run_generic_operation('Restart Servers', ems_ref)
+      end
+
+      def reload_middleware_server_group(ems_ref)
+        run_generic_operation('Reload Servers', ems_ref)
+      end
+
+      def suspend_middleware_server_group(ems_ref, params = {})
+        run_generic_operation('Suspend Servers', ems_ref, :timeout => params[:timeout] || 0)
+      end
+
+      def resume_middleware_server_group(ems_ref)
+        run_generic_operation('Resume Servers', ems_ref)
+      end
+
+      def create_jdr_report(ems_ref)
+        run_generic_operation(:JDR, ems_ref)
+      end
+
+      def remove_middleware_datasource(ems_ref)
+        run_generic_operation(operation_name, ems_ref)
+      end
+
+      def run_generic_operation(operation_name, ems_ref, params = {})
+        GenericServerOperation
+          .new(self, operation_name)
+          .run_generic(ems_ref, params)
```

Comment:
perhaps I am blind, but where is the `run_generic` method defined? I can see only the `run` method in the `GenericServerOperation`